client/deployments/client.go

```diff
@@ -296,6 +297,24 @@ func (c *Client) DirectUpload(
 		return errors.Wrap(err, "Cannot read artifact file stats")
 	}
 
+	tr := tar.NewReader(artifact)
+	versionH, err := tr.Next()
+	if err != nil {
+		return errors.Wrap(err, "Cannot find artifact")
+	} else if versionH.Name != "version" {
+		return errors.New("Invalid artifact format")
+	}
+	v := struct {
+		Format string `json:"format"`
+	}{}
+	err = json.NewDecoder(tr).Decode(&v)
+	if err != nil || v.Format != "mender" {
+		return errors.New("Invalid artifact format")
+	}
+	_, err = artifact.Seek(0, io.SeekStart)
+	if err != nil || v.Format != "mender" {
+		return err
+	}
```

Comment:
Nitpick: Can you refactor this artifact parsing part into it's own function and reuse it below?
Duplicating code often leads to forgetting about the existence about the other instance when touching the code later.